src/index.ts

```diff
@@ -52,11 +52,14 @@ const plugins = kernel_list.map((kernel): JupyterLiteServerPlugin<void> => {
         );
       }
 
+      const contentsManager = app.serviceManager.contents;
+
       kernelspecs.register({
         spec: kernelspec,
         create: async (options: IKernel.IOptions): Promise<IKernel> => {
           const mountDrive = !!(
-            serviceWorker?.enabled && broadcastChannel?.enabled
+            (serviceWorker?.enabled && broadcastChannel?.enabled) ||
```

Comment:
As discussed in https://github.com/jupyterlite/jupyterlite/issues/1294#issuecomment-1900008489, we'll probably still want to support the service worker, for deployments that are not able set the proper server headers, or don't want to.

Leaving a comment here so we can document this somewhere, in this repo or in the JupyterLite repo directly.

Reply:
Definitely, the changes here are doing exactly this.

If `crossOriginIsolated` is true, it will use `Atomics.wait` for the blocking calls, if not and the service worker is available, it will use the service worker.

Reply:
Sounds good :+1: